pupil_src/shared_modules/video_capture/ndsi_backend.py

```diff
@@ -408,76 +433,83 @@ def re_build_ndsi_menu(self):
-
-        def view_host(host_uuid):
-            if self.selected_host != host_uuid:
-                self.selected_host = host_uuid
-                self.re_build_ndsi_menu()
-
-        host_sel, host_sel_labels = host_selection_list()
+        host_sel, host_sel_labels = self.host_selection_list()
         ui_elements.append(
             ui.Selector(
                 "selected_host",
                 self,
                 selection=host_sel,
                 labels=host_sel_labels,
-                setter=view_host,
+                setter=self.view_host,
                 label="Remote host",
             )
         )
 
         self.menu.extend(ui_elements)
+
+        self.menu.append(
+            ui.Button("Auto select in all processes", self.auto_select_manager)
+        )
+
         if not self.selected_host:
             return
         ui_elements = []
 
         host_menu = ui.Growing_Menu("Remote Host Information")
         ui_elements.append(host_menu)
 
-        def source_selection_list():
-            default = (None, "Select to activate")
-            # self.poll_events()
-            sources = [default] + [
-                (s["sensor_uuid"], s["sensor_name"])
-                for s in self.network.sensors.values()
-                if (
-                    s["sensor_type"] == "video" and s["host_uuid"] == self.selected_host
-                )
-            ]
-            return zip(*sources)
-
-        def activate(source_uid):
-            if not source_uid:
-                return
-            settings = {
-                "frame_size": self.g_pool.capture.frame_size,
-                "frame_rate": self.g_pool.capture.frame_rate,
-                "source_id": source_uid,
-            }
-            self.activate_source(settings)
-
-        src_sel, src_sel_labels = source_selection_list()
+        src_sel, src_sel_labels = self.source_selection_list()
         host_menu.append(
             ui.Selector(
                 "selected_source",
                 selection=src_sel,
                 labels=src_sel_labels,
                 getter=lambda: None,
-                setter=activate,
+                setter=self.activate,
                 label="Source",
             )
         )
 
         self.menu.extend(ui_elements)
 
+    def activate(self, source_uid):
+        if not source_uid:
+            return
+        settings = {
+            "frame_size": self.g_pool.capture.frame_size,
+            "frame_rate": self.g_pool.capture.frame_rate,
+            "source_id": source_uid,
+        }
+        self.activate_source(settings)
+
+    def auto_select_manager(self):
+        super().auto_select_manager()
+        self.notify_all(
+            {
+                "subject": "notify.selected_host",
+                "selected_host": self.selected_host,
+                "delay": 0.4,
+            }
+        )
+
+    def auto_activate(self):
+        cam_selection_lut = {"eye0": "ID0", "eye1": "ID1", "world": "ID2"}
```

Comment:
This could be a class attribute for efficiency.

Reply:
My train of thought was "this function is to be seldomly called, so it's better not the bloating the memory up".

But you're right, we should favor performance over memory.